app/importers/reading/mega_reading_processor.rb

```diff
@@ -19,12 +19,14 @@ def initialize(educator_id, options = {})
     reset_counters!
   end
 
-  def import(file_text, options = {})
+  def process(file_text, options = {})
     reset_counters!
 
     # parse
     rows = []
-    StreamingCsvTransformer.from_text(@log, file_text).each_with_index do |row, index|
+    #last row before data is the header
+    string = (@header_rows_count > 1) ? file_text.lines[@header_rows_count-1..-1].join : file_text
```

Comment:
Can you say more about moving this before the CSV is parsed, here rather than afterward in `flat_map_rows`?

Reply:
Sure thing. `StreamingCSVTransformer` needs a string that can be parsed as a csv. This is here to trim header rows that we aren't going to use later to reference data points, assuming here the last line before the data begins is the "real" header. I haven't been able to find a cleaner way to specify which row in the csv to treat as the header. `flat_map_rows` just takes a csv row as an argument. It needs to reference the header data. 

An alternative approach would be to define the headers explicitly in code rather than trying to get them from the csv.

Reply:
Ah got it, thanks!  👍 Yeah so the problem is that in that template, the actual "rows" in the second row since the first row has other stuff helpful for people entering data, got it.

The new reading template format reverses that - the first row is hidden from people, so it makes for simple parsing, and then there some formats have a a few rows without any real data, with explanations for people about how to use the sheet, etc.

So this makes total sense as a workaround, and then we should be able to remove it with the new format.

Reply:
That's great! Since this is likely to be sensitive to formatting changes, there are probably a few pieces here that might need to change with new formats. That shouldn't be a problem as long as the final template works well for everyone using it.

Reply:
Yep, let me know if you want to chat more.  It's possible there will be some drift in the format from now until September, but I'm guessing it'll just be things like names to columns, or maybe skipping a different number of header rows or something like that.